lambda_functions/va_profile/va_profile_opt_in_out_lambda.py

```diff
@@ -448,6 +493,157 @@ def make_PUT_request(
+                'personalisation': {'month-name': month_personalisation},
+            }
+        )
+
+        logger.debug('Sending Comp and Pen opt-in confirmation SMS notification vaProfileId %s', va_profile_id)
+
+        conn = HTTPSConnection('{}.api.notifications.va.gov'.format(NOTIFY_ENVIRONMENT), context=ssl_context)
+        encoded_header = generate_jwt(COMP_AND_PEN_OPT_IN_API_KEY, COMP_AND_PEN_SERVICE_ID)
+
+        conn.request(
+            'POST',
+            '/v2/notifications/sms',
+            body=sms_data,
+            headers={'Content-Type': 'application/json', 'Authorization': 'Bearer {}'.format(encoded_header)},
+        )
+
+        response = conn.getresponse()
+
+        if response.status == 201:
+            logger.info(
+                'Comp and Pen opt-in confirmation SMS notification sent successfully. Response status: %s',
+                {response.status},
+            )
+            return response
+        else:
+            logger.error(
+                'Failed to send Comp and Pen opt-in confirmation SMS notification. Response status: %s',
+                {response.status},
+            )
+
+    except ValueError as ve:
+        logger.exception(
+            'Configuration error while attempting to send Comp and Pen opt-in confirmation SMS notification: %s', {ve}
+        )
+
+    except Exception as e:
+        logger.exception(
+            'An error occurred while attempting to send Comp and Pen opt-in confirmation SMS notification to vaProfileId %s: %s',
+            va_profile_id,
+            e,
+        )
+    return None
+
+
+def save_notification_id_to_cache(va_profile_id: int, notification_id: str, source_date: str):
+    """
+    Update the VAProfileLocalCache table by inserting the notification_id for the given va_profile_id.
+
+    Args:
+        va_profile_id (int): The VA profile ID of the user the notification was sent to.
+        notification_id (str): The notification UUID.
+    """
+    try:
+        with db_connection.cursor() as cursor:
+            # Execute the SQL query with the provided parameters.
+            cursor.execute(OPT_IN_OUT_ADD_NOTIFICATION_ID_QUERY, (notification_id, va_profile_id, source_date))
+
+            db_connection.commit()
+            logger.info(
+                'Successfully updated VAProfileLocalCache with notification_id %s for va_profile_id %s.',
+                notification_id,
+                va_profile_id,
+            )
+
+    except psycopg2.IntegrityError as e:
+        db_connection.rollback()
+        logger.error(
+            'Failed to insert notification_id %s for va_profile_id %s. IntegrityError: %s',
+            notification_id,
+            va_profile_id,
+            str(e),
+        )
+    except Exception as e:
+        db_connection.rollback()
+        logger.error(
+            'An error occurred while attempting to update notification_id for va_profile_id %s: %s',
```

Comment:
we should include the notification_id in the log since we have it

Reply:
Updated.